app/helpers/concerns/foreman_bootdisk/settings_helper_extensions.rb

```diff
@@ -0,0 +1,29 @@
+module ForemanBootdisk
+  module SettingsHelperExtensions
+    extend ActiveSupport::Concern
+
+    included do
+      alias_method_chain :value, :bootdisk
+    end
+
+    def value_with_bootdisk(setting)
+      return value_without_bootdisk(setting) unless [
+        'bootdisk_generic_host_template',
+        'bootdisk_host_template'].include?(setting.name)
+
+      case setting.name
+      when "bootdisk_generic_host_template"
```

Comment:
It appears that https://github.com/theforeman/foreman/commit/ad603e4023c5400c51674f1fa3635565f7b3b552#diff-8c3f2a547e7f7d66e1756971a1ef20c2R275 adds some support for passing a collection into the creation of Settings, so there's no need to extend Foreman's helper here. This could be moved to the Setting::Bootdisk class.